app/serializers/site_serializer.rb

```diff
@@ -288,23 +288,15 @@ def include_denied_emojis?
   end
 
   def tos_url
-    if SiteSetting.tos_url.present?
-      SiteSetting.tos_url
-    elsif SiteSetting.tos_topic_id > 0 && Topic.exists?(id: SiteSetting.tos_topic_id)
-      "#{Discourse.base_path}/tos"
-    end
+    Discourse.tos_path
```

Comment:
Note that each time `Discourse.tos_path` is called, we're reading from Redis so we need to memoize this. Alternatively, we can consider using a `DistributedCache` instead where fetching from the cache is just a hash lookup.